lib/fs.js

```diff
@@ -1621,8 +1623,12 @@ function ReadStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    internalUtil.printDeprecationMessage('Passing anything but an object or \
```

Comment:
This message will be printed whenever this situation arises. Is that what is intended? Normally, it is used like this

    var warned = false;
    ...
        warned = internalUtil.printDeprecationMessage('...', warned);